server/src/main/java/org/elasticsearch/search/aggregations/support/AggregationContext.java

```diff
@@ -435,7 +435,7 @@ public Analyzer buildCustomAnalyzer(
 
         @Override
         protected IndexFieldData<?> buildFieldData(MappedFieldType ft) {
-            return context.getForField(ft);
+            return context.getForField(ft, MappedFieldType.FielddataOperation.SEARCH);
```

Comment:
given all these related changes, too late now but have we considered having a separate getForField only for scripting and leaving the existing one alone, so that less changes are required in existing code?